enumerables.rb

```diff
@@ -0,0 +1,199 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Style/RedundantSelf, Style/GuardClause, Style/IfUnlessModifier, Metrics/BlockNesting, Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    array = self if self.class == Array
+    array = self.to_a if self.class == Range
+    i = 0
+    while i < array.length
+      if block_given?
+        yield array[i]
+      else
+        return "\#<Enumerator: #{self}:my_each>"
+      end
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    array = self if self.class == Array
+    array = self.to_a if self.class == Range
+    value = 0
+    while value < array.length
+      if block_given?
+        yield(array[value], value)
+      else
+        return "\#<Enumerator: #{self}:my_each_with_index>"
+      end
+      value += 1
+    end
+    array
+  end
+
+  def my_select
+    array = []
+    i = 0
+    while i < self.length
+      if yield self[i]
+        array.push(self[i])
+      end
+      i += 1
+    end
+    array
+  end
+
+  def my_all?(*args)
+    if args.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield(self[i]) == false
+            return false
+          end
+        elsif self[i].nil?
+          return false
+        else
+          return true
+        end
+        i += 1
+      end
+    else
+      return false
+    end
+    true
+  end
```

Comment:
- [ ] my_all? returns false if the block returns false or nil in a range

- [ ] when a class is passed as an argument it returns true if all of the collection is a member of such class

- [ ] when a Regex is passed as an argument it returns true if all of the collection matches the Regex
ex: (words.my_all?(/d/)) should be equeal words.all?(/d/)

- [ ] when a pattern other than Regex or a Class is given it returns true if all of the collection matches the pattern
ex: (array.my_all?(3)) should be equal array.all?(3)